### PR TITLE
Added Solidity -> ink! guide

### PR DESCRIPTION
**Pull request by peterwht**
Added a guide to convert from Solidity to ink! contracts. This is placed in the ink-vs-solidity page. Original guide can be found here: https://github.com/peterwht/solidity-to-ink-guide.

### CODE REVIEW

File: docs/intro/ink-vs-solidity.md [hunk 1]
```diff
@@ -7,19 +7,929 @@ Here is a brief comparison of features between ink! and Solidity:
...
+  - [`cross-contract calling`](#cross-contract-calling)
+  - [`submit generic transaction / dynamic cross-contract calling`](#submit-generic-transaction--dynamic-cross-contract-calling)
+- [Limitations of ink! v3](#limitations-of-ink-v3)
+- [Troubleshooting Errors](#troubleshooting-errors)
+- [unit testing (off-chain)](#unit-testing-off-chain)
+
+## What is ink!
+
+ink! is the smart contract language used in Substrate. It is built from Rust -- meaning that the great features of Rust are included in ink!. These features are ideal for a smart contract language. Furthermore, ink! is compiled to WebAssembly, which is high-performant, consistent, and very well researched. Furthermore, using Rust allows for inherent [safe math](https://ink.substrate.io/faq#overflow-safety) checks. ink! smart contracts use Rust's no_std. So, some common Rust implementations are not directly supported. However, ink! does have crates that reimplement common Rust code to work in the Substrate runtime.
+
```
Comment (cmichi) [inline docs/intro/ink-vs-solidity.md:71]:
```suggestion
## What is ink!

ink! is the smart contract language used in Substrate. It is built from Rust -- meaning that the great features of Rust are included in ink!. These features are ideal for a smart contract language. Furthermore, ink! is compiled to WebAssembly, which is high-performant, consistent, and very well researched. Furthermore, using Rust allows for inherent [safe math](https://ink.substrate.io/faq#overflow-safety) checks. ink! smart contracts use Rust's no_std. So, some common Rust implementations are not directly supported. However, ink! does have crates that reimplement common Rust code to work in the Substrate runtime.
```

Can be left out, as we're already in the ink! docs here.

File: docs/intro/ink-vs-solidity.md [hunk 2]
```diff
@@ -7,19 +7,929 @@ Here is a brief comparison of features between ink! and Solidity:
...
+## Setup
+
+Follow the [prepare-your-first-contract](https://docs.substrate.io/tutorials/smart-contracts/prepare-your-first-contract/) tutorial.
```
Comment (cmichi) [inline docs/intro/ink-vs-solidity.md:74]:
```suggestion
```

File: docs/intro/ink-vs-solidity.md [hunk 3]
```diff
@@ -7,19 +7,929 @@ Here is a brief comparison of features between ink! and Solidity:
...
+
+## Converting Solidity Contract to ink!
```
Comment (cmichi) [inline docs/intro/ink-vs-solidity.md:76]:
```suggestion
## Converting a Solidity Contract to ink!
```

File: docs/intro/ink-vs-solidity.md [hunk 4]
```diff
@@ -7,19 +7,929 @@ Here is a brief comparison of features between ink! and Solidity:
...
+
+### 1. Generate New ink! Contract
+
+Run the following to generate ink! boilerplate code (flipper project)
```
Comment (cmichi) [inline docs/intro/ink-vs-solidity.md:80]:
```suggestion
Run the following to generate ink! boilerplate code for ink!'s "Hello, World!" (the [`flipper`](https://github.com/paritytech/ink/tree/master/examples/flipper) contract))
```

File: docs/intro/ink-vs-solidity.md [hunk 5]
```diff
@@ -484,7 +474,9 @@ assert!(caller == owner, "caller is not owner")
...
+In Solidity, `require` is used for general (normal) errors -- such as errors that occur based on user input. `require` does have the option for an error string. `revert` is very similar to `require` except that `revert` will be called in `if ... else` chains. Both `require` and `revert` will revert the chain state. In ink!, `if ... { return Err(Error::SomeError) }` should be used for `require` or `revert`. When a `Result::Err` is returned in ink!, then all state is reverted.
+
+In general, `Result::Err` should be used when a _calling contract_ needs to know _why_ a function failed. Otherwise, `assert!` should be used as it has less overhead than a `Result`.
```
Comment (peterwht) [inline docs/intro/ink-vs-solidity.md:479]: Is the last sentence accurate? Based on the discussion [here](https://github.com/paritytech/ink/issues/1335).

**Review by cmichi — APPROVED** [review]
Thank you, Peter!

**Pull request merged**